packages/devtools-proxy-support/src/system-ca.ts

```diff
@@ -50,6 +51,54 @@ export function mergeCA(...args: (NodeJSCAOption | undefined)[]): string {
   return [...ca].join('\n');
 }
 
+const pemWithParsedCache = new WeakMap<
+  string[],
+  { pem: string; parsed: X509Certificate | null }[]
+>();
+// TODO(COMPASS-8253): Remove this in favor of OpenSSL's X509_V_FLAG_PARTIAL_CHAIN
+// See linked tickets for details on why we need this (tl;dr: the system certificate
+// store may contain intermediate certficiates without the corresponding trusted root,
+// and OpenSSL does not seem to accept that)
+export function removeCertificatesWithoutIssuer(ca: string[]): {
+  ca: string[];
+  messages: string[];
+} {
+  const messages: string[] = [];
+  let caWithParsedCerts =
+    pemWithParsedCache.get(ca) ??
+    ca.map((pem) => {
+      let parsed: X509Certificate | null = null;
+      try {
+        parsed = new X509Certificate(pem);
+      } catch (err: unknown) {
+        messages.push(
+          `Unable to parse certificate: ${
+            err && typeof err === 'object' && 'message' in err
+              ? String(err.message)
+              : String(err)
+          }`
+        );
+      }
+      return { pem, parsed };
+    });
+  pemWithParsedCache.set(ca, caWithParsedCerts);
+  caWithParsedCerts = caWithParsedCerts.filter(({ parsed }) => {
+    const keep =
+      !parsed ||
```

Comment:
This is somewhat counterintuitive to me - if we were unable to parse the certificate, do we want to keep it? I guess there's no harm in doing it, but do we feel it's useful to hold on to pems that are not valid x509 certificates?

Reply:
I don't think we should ever encounter a situation like this in the first place. But if we do, I think this is the "safer" path – if nothing else, it aligns more closely with the previous behavior (and the one we're aiming for in the long term, once this code is removed again). So I think just leaving a diagnostic message here is probably okay?